smarts/core/opendrive_road_network.py

```diff
@@ -0,0 +1,1810 @@
+                :leftmost_edge_vertices_len
+            ]
+
+            return leftmost_edge_shape, rightmost_edge_shape
+
+        @lru_cache(maxsize=4)
+        def shape(self, width: float = 0.0, buffer_width: float = 0.0) -> Polygon:
+            leftmost_edge_shape, rightmost_edge_shape = self._shape(buffer_width)
+            road_polygon = (
+                leftmost_edge_shape + rightmost_edge_shape + [leftmost_edge_shape[0]]
+            )
+            return Polygon(road_polygon)
+
+        def lane_at_index(self, index: int) -> RoadMap.Lane:
+            lanes_with_index = [lane for lane in self.lanes if lane.index == index]
+            if len(lanes_with_index) == 0:
+                self._log.warning(
+                    f"Road with id {self.road_id} has no lane at index {index}"
+                )
+                return None
+            assert len(lanes_with_index) == 1
+            return lanes_with_index[0]
+
+    def road_by_id(self, road_id: str) -> RoadMap.Road:
+        road = self._roads.get(road_id)
+        if not road:
+            self._log.warning(
+                f"OpenDriveRoadNetwork got request for unknown road_id '{road_id}'"
+            )
+        return road
+
+    def _build_lane_r_tree(self):
+        result = rtree.index.Index()
+        result.interleaved = True
+        for idx, lane in enumerate(self._all_lanes):
+            bounding_box = (
+                lane.bounding_box[0][0],
+                lane.bounding_box[0][1],
+                lane.bounding_box[1][0],
+                lane.bounding_box[1][1],
+            )
+            result.add(idx, bounding_box)
+        return result
+
+    def _get_neighboring_lanes(self, x, y, r=0.1):
+        neighboring_lanes = []
+
+        if self._lane_rtree is None:
+            self._lane_rtree = self._build_lane_r_tree()
+
+        assert len(self._all_lanes) != 0
+        for i in self._lane_rtree.intersection((x - r, y - r, x + r, y + r)):
+            lane = self._all_lanes[i]
+            d = distance_point_to_polygon((x, y), lane.lane_polygon)
+            if d < r:
+                neighboring_lanes.append((lane, d))
+        return neighboring_lanes
+
+    @lru_cache(maxsize=16)
+    def nearest_lanes(
+        self, point: Point, radius: float = None, include_junctions=True
+    ) -> List[Tuple[RoadMap.Lane, float]]:
+        if radius is None:
+            radius = max(10, 2 * self._default_lane_width)
+        candidate_lanes = self._get_neighboring_lanes(point[0], point[1], r=radius)
+        candidate_lanes.sort(key=lambda lane_dist_tup: lane_dist_tup[1])
+        return candidate_lanes
+
+    def nearest_lane(
+        self, point: Point, radius: float = None, include_junctions=True
+    ) -> RoadMap.Lane:
+        nearest_lanes = self.nearest_lanes(point, radius, include_junctions)
+        for lane, dist in nearest_lanes:
+            if lane.contains_point(point):
+                return lane
```

Comment:
`dist` here is the distance from the central line of a given lane. Since `OpenDRIVE` has lanes of varying width, a point can be closer to a lane it does not lie in when compared to the lane it does if it is more closer to the outer lane's central line, than the lane it lies in.

Reply:
ah ha, makes sense!  thanks!
Given that it's not obvious, you should probably add a short comment explaining this in the `nearest_lane` method.